src/shared/bus-polkit.c

```diff
@@ -323,7 +323,8 @@ static int async_polkit_read_reply(sd_bus_message *reply, AsyncPolkitQuery *q) {
         } else if (challenge) {
                 log_debug("Polkit authorization for action '%s' requires interactive authentication, which we didn't allow.", a->action);
                 q->error_action = TAKE_PTR(a);
-                sd_bus_error_set_const(&q->error, SD_BUS_ERROR_INTERACTIVE_AUTHORIZATION_REQUIRED, "Interactive authentication required.");
+                sd_bus_error_set_const(&q->error, SD_BUS_ERROR_INTERACTIVE_AUTHORIZATION_REQUIRED,
+                                       "Access denied as the requested operation requires interactive authentication. However, interactive authentication has been disabled by the client.");
```

Comment:
This is probably bikeshedding but since its a user facing error:

- I personally only partially understand what "client" means in this context (presumably the system?) and I would think a different less technical term is something more people would be able to understand (e.g. `system` or `environment`?), either way "client" is probably too vague.
- "has been" implies to me some deliberate change that is unmodifyable, but since this isn't in our control we probably should assume that it is possibly changable and might not be deliberate.

```suggestion
                                       "Access denied as the requested operation requires interactive authentication. However, interactive authentication is disabled on the system.");
```

Reply:
This is a misunderstanding. Polkit is IPC-based, and client means the IPC client here, i.e. the tool or app that does the IPC called.

And yes, it's precisely that client that turned off interactive authentication for the IPC call it is doing.

Reply:
I guess we could change client to "caller". Would that make this clearer?

Reply:
or maybe "calling program"

Reply:
> Polkit is IPC-based, and client means the IPC client here

Then I just partially misunderstood how exactly polkit auth is set up (I did know it was via IPC, but unsure what decides what level of auth is needed), but that is whatever.

"Calling program" does sound fine to me, I don't have anything that would be easier to understand.